### PR TITLE
GH-41696: [Python][Packaging] Bump MACOSX_DEPLOYMENT_TARGET to 12 instead of 11

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

As shown on the associated issue there seems to be a problem with `MACOSX_DEPLOYMENT_TARGET` 11 on the wheels.

### What changes are included in this PR?

Update `MACOSX_DEPLOYMENT_TARGET` everywhere to the latest supported macOS version.

### Are these changes tested?

Via CI, even though the issue was not reproducible on CI.

### Are there any user-facing changes?

Yes, wheels won't be available for macOS 11 but those were crashing on the previous release.
* GitHub Issue: #41696

### CODE REVIEW

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-*

**Comment by raulcd** [comment]
@github-actions crossbow submit verify-rc-source-*-macos-*

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-monterey-cp310-amd64

**Comment by raulcd** [comment]
I am not sure why the linking error is happening:
```
[125/133] Linking CXX shared library release/libarrow_flight.1700.0.0.dylib
FAILED: release/libarrow_flight.1700.0.0.dylib 
: && /Applications/Xcode_15.0.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -fno-aligned-new  -Qunused-arguments -fcolor-diagnostics  -Wall -Wno-unknown-warning-option -Wno-pass-failed -msse4.2  -O3 -DNDEBUG -O2 -arch x86_64 -isysroot /Applications/Xcode_15.0.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -mmacosx-version-min=12.0 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup   -compatibility_version 1700.0.0 -current_version 1700.0.0 -o release/libarrow_flight.1700.0.0.dylib -install_name @rpath/libarrow_flight.1700.dylib src/arrow/flight/CMakeFiles/arrow_flight_objlib.dir/Unity/unity_2_cxx.cxx.o src/arrow/flight/CMakeFiles/arrow_flight_objlib.dir/Unity/unity_1_cxx.cxx.o src/arrow/flight/CMakeFiles/arrow_flight_objlib.dir/Unity/unity_0_cxx.cxx.o  -Wl,-rpath,@loader_path  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libgrpc++.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd
ld: imageOffset32 fixup target must be defined
```

**Comment by vserdyuk** [comment]
> I am not sure why the linking error is happening:
> 
> ```
> FAILED: release/libarrow_flight.1700.0.0.dylib 
> : && /Applications/Xcode_15.0.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -fno-aligned-new  -Qunused-arguments -fcolor-diagnostics  -Wall -Wno-unknown-warning-option -Wno-pass-failed -msse4.2  -O3 -DNDEBUG -O2 -arch x86_64 -isysroot /Applications/Xcode_15.0.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -mmacosx-version-min=12.0 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup   -compatibility_version 1700.0.0 -current_version 1700.0.0 -o release/libarrow_flight.1700.0.0.dylib -install_name @rpath/libarrow_flight.1700.dylib src/arrow/flight/CMakeFiles/arrow_flight_objlib.dir/Unity/unity_2_cxx.cxx.o src/arrow/flight/CMakeFiles/arrow_flight_objlib.dir/Unity/unity_1_cxx.cxx.o src/arrow/flight/CMakeFiles/arrow_flight_objlib.dir/Unity/unity_0_cxx.cxx.o  -Wl,-rpath,@loader_path  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libgrpc++.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libprotobuf.a  release/libarrow.1700.0.0.dylib  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libgrpc.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libz.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libcares.a  -lresolv  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libaddress_sorting.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libre2.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_internal.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_marshalling.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_reflection.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_config.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_program_name.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_private_handle_accessor.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_commandlineflag.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_commandlineflag_internal.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_raw_hash_set.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_hashtablez_sampler.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_hash.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_city.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_low_level_hash.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_statusor.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libgpr.a  -lm  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_json.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_textformat.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_reflection.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_collections.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_mini_table.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_fastdecode.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_utf8_range.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_extension_registry.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libdescriptor_upb_proto.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_bad_variant_access.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_status.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_strerror.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_distributions.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_seed_sequences.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_internal_pool_urbg.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_internal_randen.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_internal_randen_hwaes.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_internal_randen_hwaes_impl.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_internal_randen_slow.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_internal_platform.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_internal_seed_material.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_seed_gen_exception.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_cord.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_bad_optional_access.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_cordz_info.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_cord_internal.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_cordz_functions.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_exponential_biased.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_cordz_handle.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_synchronization.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_graphcycles_internal.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_kernel_timeout_internal.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_stacktrace.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_symbolize.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_malloc_internal.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_debugging_internal.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_demangle_internal.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_crc_cord_state.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_crc32c.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_str_format_internal.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_crc_internal.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_crc_cpu_detect.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_time.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_strings.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_strings_internal.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_string_view.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_base.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_spinlock_wait.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_throw_delegate.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_int128.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_raw_logging_internal.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_log_severity.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_civil_time.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_time_zone.a  -Wl,-framework,CoreFoundation  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libssl.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libcrypto.a  -framework CoreFoundation && :
> ld: imageOffset32 fixup target must be defined
> ```

I have the same error when running vcpkg install qtbase:x64-osx-dynamic. Seems like Xcode toolchain issue.

**Comment by jorisvandenbossche** [comment]
@raulcd does it work if we leave the deployment target in the vcpkg files as is? (in the end for pypi what matters is the macos version that ends up in the wheels, so that older macos don't think they can install the wheel)

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-monterey-cp310-amd64

**Comment by raulcd** [comment]
Try reverting vpkg changes as suggested

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-monterey-cp310-amd64

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-catalina-cp310-amd64

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-monterey-cp310-amd64

**Comment by raulcd** [comment]
It seems that as soon as we update the macos deployment target to 12.0  we get the flight linking issue:
```
FAILED: release/libarrow_flight.1800.0.0.dylib 
: && /Applications/Xcode_15.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -fno-aligned-new  -Qunused-arguments -fcolor-diagnostics  -Wall -Wno-unknown-warning-option -Wno-pass-failed -msse4.2  -O3 -DNDEBUG -O2 -arch x86_64 -isysroot /Applications/Xcode_15.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk -mmacosx-version-min=12.0 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup   -compatibility_version 1800.0.0 -current_version 1800.0.0 -o release/libarrow_flight.1800.0.0.dylib -install_name @rpath/libarrow_flight.1800.dylib src/arrow/flight/CMakeFiles/arrow_flight_objlib.dir/Unity/unity_2_cxx.cxx.o src/arrow/flight/CMakeFiles/arrow_flight_objlib.dir/Unity/unity_1_cxx.cxx.o src/arrow/flight/CMakeFiles/arrow_flight_objlib.dir/Unity/unity_0_cxx.cxx.o  -Wl,-rpath,@loader_path  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libgrpc++.a  /Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-o
ld: imageOffset32 fixup target must be defined
```
@kou any idea what the issue could be?

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-big-sur-cp310-amd64

**Comment by raulcd** [comment]
If I change the macos runner to use 12 (monterey) matching the `MACOSX_DEPLOYMENT_TARGET` it is able to build.
The initial issue is probably related to an issue between `Xcode_15.2` and `-mmacosx-version-min=12.0`. The job is still failing because it can't import `pyarrow.flight` due to missing `_OPENSSL_ia32cap_P` symbol:
```
python -c 'import pyarrow.flight'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/runner/work/crossbow/crossbow/test-env/lib/python3.10/site-packages/pyarrow/flight.py", line 67, in <module>
    raise ImportError(
ImportError: The pyarrow installation is not built with support for 'flight' (dlopen(/Users/runner/work/crossbow/crossbow/test-env/lib/python3.10/site-packages/pyarrow/_flight.cpython-310-darwin.so, 0x0002): symbol not found in flat namespace (_OPENSSL_ia32cap_P))
```

**Comment by raulcd** [comment]
debugging with @assignUser it seems that previously the OpenSSL symbols were defined on `libarrow_flight` where now they aren't. New wheel:
```
nm -C libarrow_flight.1800.dylib  | grep OPENSSL_ia                       
                 U _OPENSSL_ia32cap_P
```
existing wheel for 17.0.0:
```
nm -C  ~/miniforge3/envs/x86/lib/python3.10/site-packages/pyarrow/libarrow_flight.1700.dylib | grep OPENSSL_ia
0000000000553020 T _OPENSSL_ia32_cpuid
0000000000553420 T _OPENSSL_ia32_rdrand_bytes
0000000000553490 T _OPENSSL_ia32_rdseed_bytes
00000000006cf8f0 s _OPENSSL_ia32cap_P
```

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-monterey-cp310-amd64

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-monterey-cp310-amd64

File: swift/ArrowFlight/Sources/ArrowFlight/Flight.grpc.swift
Comment (kou) [inline swift/ArrowFlight/Sources/ArrowFlight/Flight.grpc.swift:1]: This is an auto-generated file. So we don't need to change this.

File: dev/tasks/tasks.yml [hunk 1]
```diff
@@ -438,7 +438,7 @@ tasks:
       arch: "arm64"
       arrow_jemalloc: "OFF"
       python_version: "{{ python_version }}"
-      macos_deployment_target: "11.0"
+      macos_deployment_target: "12.0"
```
Comment (kou) [inline dev/tasks/tasks.yml:441]: We need to update `big-sur` in `wheel-macos-big-sur-{{ python_tag }}-arm64` too.

**Review by kou — COMMENTED** [review]
Hmm. If `macos-12` ships Xcode 14.3 not 14.2, this problem may be solved...

It seems that the `_OPENSSL_ia32cap_P` problem is a general problem. Can we try https://stackoverflow.com/questions/42439834/openssl-library-linker-command-failed-openssl-ia32cap-p-in-aes-decoding#comment108622544_42439834 ?

> I also had this problem on macOS, the solution on my side was to make sure my `OTHER_LDFLAGS` had `-lssl` and `-lcrypto` before any `-framework` argument

Based on https://github.com/ursacomputing/crossbow/actions/runs/10576927316/job/29303788700#step:12:1678 , we may have `-framework CoreFoundation` after `libssl.a` and `libcrypto.a`.

```text
/Applications/Xcode_15.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++
-fno-aligned-new
-Qunused-arguments
-fcolor-diagnostics
-Wall
-Wno-unknown-warning-option
-Wno-pass-failed
-msse4.2
-O3
-DNDEBUG
-O2
-arch
x86_64
-isysroot
/Applications/Xcode_15.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk
-mmacosx-version-min=12.0
-dynamiclib
-Wl,-headerpad_max_install_names
-undefined
dynamic_lookup
-compatibility_version
1800.0.0
-current_version
1800.0.0
-o
release/libarrow_flight.1800.0.0.dylib
-install_name
@rpath/libarrow_flight.1800.dylib
src/arrow/flight/CMakeFiles/arrow_flight_objlib.dir/Unity/unity_2_cxx.cxx.o
src/arrow/flight/CMakeFiles/arrow_flight_objlib.dir/Unity/unity_1_cxx.cxx.o
src/arrow/flight/CMakeFiles/arrow_flight_objlib.dir/Unity/unity_0_cxx.cxx.o
-Wl,-rpath,@loader_path
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libgrpc++.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libprotobuf.a
release/libarrow.1800.0.0.dylib
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libgrpc.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libz.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libcares.a
-lresolv
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libaddress_sorting.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libre2.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_internal.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_marshalling.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_reflection.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_config.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_program_name.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_private_handle_accessor.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_commandlineflag.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_flags_commandlineflag_internal.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_raw_hash_set.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_hashtablez_sampler.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_hash.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_city.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_low_level_hash.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_statusor.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libgpr.a
-lm
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_json.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_textformat.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_reflection.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_collections.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_mini_table.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_fastdecode.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_utf8_range.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libupb_extension_registry.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libdescriptor_upb_proto.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_bad_variant_access.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_status.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_strerror.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_distributions.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_seed_sequences.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_internal_pool_urbg.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_internal_randen.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_internal_randen_hwaes.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_internal_randen_hwaes_impl.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_internal_randen_slow.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_internal_platform.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_internal_seed_material.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_random_seed_gen_exception.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_cord.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_bad_optional_access.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_cordz_info.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_cord_internal.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_cordz_functions.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_exponential_biased.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_cordz_handle.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_synchronization.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_graphcycles_internal.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_kernel_timeout_internal.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_stacktrace.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_symbolize.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_malloc_internal.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_debugging_internal.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_demangle_internal.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_crc_cord_state.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_crc32c.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_str_format_internal.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_crc_internal.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_crc_cpu_detect.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_time.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_strings.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_strings_internal.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_string_view.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_base.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_spinlock_wait.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_throw_delegate.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_int128.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_raw_logging_internal.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_log_severity.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_civil_time.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libabsl_time_zone.a
-Wl,-framework,CoreFoundation
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libssl.a
/Users/runner/work/crossbow/crossbow/vcpkg/installed/amd64-osx-static-release/lib/libcrypto.a
-framework
CoreFoundation
```

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-monterey-cp310-amd64

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-*

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-*-arm64

**Comment by raulcd** [comment]
@kou the issue gets fixed if I explicitly link flight to OpenSSL, do you think there is a reason we were not explicitly linking? I am confused on why this wasn't the case

File: cpp/src/arrow/flight/CMakeLists.txt [hunk 2]
```diff
@@ -26,6 +26,9 @@ endif()
 if(WIN32)
   list(APPEND ARROW_FLIGHT_LINK_LIBS ws2_32.lib)
 endif()
+if(ARROW_USE_OPENSSL)
```
Comment (kou) [inline cpp/src/arrow/flight/CMakeLists.txt:29]: `if(APPLE AND ARROW_USE_OPENSSL)` may be better because this is just needed on macOS.

**Review by kou — APPROVED** [review]
+1

> the issue gets fixed if I explicitly link flight to OpenSSL, do you think there is a reason we were not explicitly linking? I am confused on why this wasn't the case

Yes. In general, we don't need to link to OpenSSL because Flight doesn't use OpenSSL API directly (no `#include <openssl/*.h>`) and gRPC also doesn't export OpenSSL API (`grpcpp/*.h` doesn't have `#include <openssl/*.h>`).

I'm not sure what has the cause of this (CoreFoundation? OpenSSL? linker?) but I think that this is the same workaround of https://github.com/apache/arrow/pull/43137#pullrequestreview-2264458735 .

Could you add a comment that this is a workaround to the following part?

```cmake
if(ARROW_USE_OPENSSL)
  list(APPEND ARROW_FLIGHT_LINK_LIBS ${ARROW_OPENSSL_LIBS})
endif()
```

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-macos-monterey-cp310-amd64

**Review by jorisvandenbossche — APPROVED** [review]
Thanks for looking into this!

**Pull request merged**